ads/gumgum.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+import {setStyles} from '../src/style';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function gumgum(global, data) {
+  validateData(data, ['zone', 'slot']);
+
+  const
+    win = window,
+    ctx = win.context,
+    dom = global.document.getElementById('c'),
+    ampWidth = data.width,
```

Comment:
data.width and data.height can be undefined with certain layout type. Please add check to avoid calling `Math.max(undefined, somenumber)`